e2e/test/BulkOperationsE2ETests.cs

```diff
@@ -0,0 +1,105 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Microsoft.Azure.Devices.Shared;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.Tracing;
+using System.Threading.Tasks;
+
+namespace Microsoft.Azure.Devices.E2ETests
+{
+    [TestClass]
+    [TestCategory("IoTHub-E2E")]
+    public class BulkOperationsE2ETests : IDisposable
+    {
+        private readonly string DevicePrefix = $"E2E_{nameof(BulkOperationsE2ETests)}_";
+        private readonly string ModulePrefix = $"E2E_{nameof(BulkOperationsE2ETests)}_";
+
+        public BulkOperationsE2ETests() {}
+
+        [TestMethod]
+        public async Task BulkOperations_UpdateTwins2Device_Ok()
+        {
+            var tagName = Guid.NewGuid().ToString();
+            var tagValue = Guid.NewGuid().ToString();
+
+            TestDevice testDevice = await TestDevice.GetTestDeviceAsync(DevicePrefix).ConfigureAwait(false);
+
+            using (RegistryManager registryManager = RegistryManager.CreateFromConnectionString(Configuration.IoTHub.ConnectionString))
+            {
+                Twin twin = await registryManager.GetTwinAsync(testDevice.Id).ConfigureAwait(false);
```

Comment:
Isn't the code fix below also populating now ModuleId in the twin obtained here? We should then have some asserts that the twin deviceId (and below moduleId) match the ones you expect (TestDevice.Id / TestModule.Id).

Reply:
no, it's gonna be a different device for TestModule, because TestModule generates it's own device